app/code/community/Bolt/Boltpay/Model/Payment.php

```diff
@@ -547,6 +557,148 @@ public function handleTransactionUpdate(Mage_Payment_Model_Info $payment, $newTr
                     $order->setState(self::ORDER_DEFERRED, true, $message);
                     $order->save();
                 }
+                elseif ($newTransactionStatus == self::TRANSACTION_REFUND) {
+                    //called by api
+                    $payment->setAdditionalInformation('bolt_transaction_refund_api', '1');
```

Comment:
you should rename this and comment it in such a way that its purpose is more self-describing. 
One possibility is:

```
// flag refund as already being set on Bolt to prevent a duplicate call by Magento to Bolt
 $payment->setAdditionalInformation('bolt_transaction_was_refunded_by_webhook', '1');
```

Reply:
agree, will do